lib/Cro/WebApp/Form.pm6

```diff
@@ -261,17 +273,22 @@ role Cro::WebApp::Form {
 
     #| Return the form data as a hash
     method form-data() {
-      my %values;
-      for self.^attributes.grep(*.has_accessor) -> Attribute $attr {
-        my $name = $attr.name.substr(2);
-        %values{$name} = $attr.get_value(self);
-      }
-      %values
+        my %values;
+        for self.^attributes.grep(*.has_accessor) -> Attribute $attr {
+            my $name = $attr.name.substr(2);
+            %values{$name} = $attr.get_value(self);
+        }
+        %values
     }
 
-    #| Take a application/x-www-form-urlencoded body and populate the form values based
+    my subset Form where Cro::HTTP::Body::WWWFormUrlEncoded | Cro::HTTP::Body::MultiPartFormData;
+
+    multi sub get-value(Cro::HTTP::Body::MultiPartFormData::Part $p) { $p.body-blob.decode('ascii') }
```

Comment:
Should this really be ASCII?

Reply:
No, but I have no idea what it should be.